***
# <p style='text-align: center;'>`Tarea 3.1`</p>
***
`Elaborada por: Lyha Espinosa García, Ana Cristina Vázquez Vargas.` `Expedientes: if721861, if721215`


1. Buy a start up stock that doesn’t pay dividends estimate the Expected Results using Corporate Finance Theory the X factor stochastic 
Assume a normal distribution.     
a)Analytical method  
b)Simulate in excel/python to test accuracy  





Haciendo uso de la fórmula vista en clase para conocer el precio justo de una acción que no paga dividendos se presenta lo siguiente:  

$$S_o=\frac{(A*F-L)}{Q}$$

Donde:  
$A$ representa los Assets (Activos)  
$F$ es conocido como el Factor de ajuste estocástico  
$L$ significa Liabilities (Pasivos)  
$Q$ es el número de acciones en circulación

$A=321195$
$L=227791$
$Q=503$


#### Para encontrar el $E[S_0]$ aplicamos las propiedades de la esperanza

$$E[S_0]=E\left [\frac{(AF-L)}{Q}  \right ]$$

$$E[S_0]=\frac{1}{Q}E\left [ AF-L \right ]$$

$$E[S_0]=\frac{1}{Q}[E\left [ AF \right ]-E\left [ L \right ]]$$

$$E[S_0]=\frac{1}{Q}[AE\left [F \right ]-E\left [ L \right ]]$$

#### Sustituyendo la información que tenemos

$$E[S_0]=\frac{1}{503}[321195E\left [F \right ]- 227791]$$

#### Asumiendo una distribución normal para nuestro factor estocástico F tenemos que:

$\mu=0$
$\sigma=1$

#### Por lo tanto

$$E[S_0]=\frac{1}{503}[321195\left [0 \right ]- 227791]$$

$$E[S_0]=\frac{1}{503}[- 227791]$$

$$E[S_0]=-452.86$$


#### Para encontrar $V[S_0]$ aplicamos las propiedades de la varianza

$$V[S_0]=V\left [ \frac{AF-L}{Q} \right ]$$

$$V[S_0]=\frac{1}{Q^{2}}V\left [ AF-L \right ]$$

$$V[S_0]=\frac{1}{Q^{2}}\left [ V[AF]-V[L] \right ]$$

$$V[S_0]=\frac{1}{Q^{2}}\left [ A^{2}V[F]-V[L] \right ]$$

$$V[S_0]=\frac{A^{2}}{Q^{2}}\left [ V[F]\right ]$$

#### Sustituyendo con la información dada

$$V[S_0]=\frac{321195^{2}}{503^{2}}\left [ 1\right ]$$

$$V[S_0]=407757.15$$

Para comprobar la certeza de nuestros cálculos analíticos, procedemos a simular el factor estocástico $F$

In [19]:
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt

In [14]:
esto=np.random.randn(1,10000)

In [9]:
A=321195
L=227791
Q=503

In [15]:
sim=[(A*esto[i]-L)/Q for i in range(len(esto))]

In [17]:
np.mean(sim)

-449.3737410769848

In [18]:
np.var(sim)

412194.28544002946

### Reporte de resultados

| Método     | $E[S_0]$  | $V[S_0]$    |
|------------|---------|-----------|
| Analítico  | −452.86 | 407757.15 |
| Simulación | -449.37 | 412194.28 |

Tal como se observa en la tabla, los valores obtenidos a traves de la simulación son similares a los obtenidos por el método analítico. Es verdad que el valor esperado de un activo no puede ser negativo, por lo que para modificar esto valdría la pena tratar los datos como una distribución distinta a la que pide el problema. 

En conclusión, el simular un proceso que se puede calcular de manera analitica, resultara en la convergencia de los resultados a valores muy similares. Por ejemplo, en esta tarea se utilizaron solo 10,000 escenarios debido a la carga computacional, no obstante, si se utilizaran 1,000,000 de escenarios, se tendría un resultado practicamente igual al obtenido por medio del método analítico. 